airflow/providers/arangodb/hooks/arangodb.py

```diff
@@ -18,14 +18,20 @@
 """This module allows connecting to a ArangoDB."""
 from __future__ import annotations
 
-from typing import Any
+from typing import TYPE_CHECKING, Any
 
 from arango import AQLQueryExecuteError, ArangoClient as ArangoDBClient
-from arango.result import Result
 
 from airflow import AirflowException
+from airflow.compat.functools import cached_property
 from airflow.hooks.base import BaseHook
 
+if TYPE_CHECKING:
```

Comment:
BTW. Why do we want to add "if TYPE_CHECKING" here @Taragolis ? 
I was under the impression that we use if TYPE_CHECKING only when we cannot do it otherwise because of importing circular references but I think in this case it is not needed ?

Reply:
That for help mypy because arangodb methods return generic type Result[{Something}] which have definition

```python
T = TypeVar("T")

Result = Union[T, AsyncJob[T], BatchJob[T], None]
```

Reply:
And actual type known only in runtime because methods wraps it into additional handlers.
In this case it returns Cursor or raise some error,  e.g.: we get [AQL.execute](https://github.com/ArangoDB-Community/python-arango/blob/1ab63c25c3b3fc152b8386c1b62c705645fb197b/arango/aql.py#L421-L424) which returns in our case [DefaultApiExecutor.execute](https://github.com/ArangoDB-Community/python-arango/blob/1ab63c25c3b3fc152b8386c1b62c705645fb197b/arango/executor.py#L55-L65)